src/utils.ts

```diff
@@ -171,8 +171,13 @@ export function applyRetryableWrites<T extends HasRetryableWrites>(target: T, db
  * @param value - An object that could be a promise
  * @returns true if the provided value is a Promise
  */
-export function isPromiseLike<T = any>(value?: PromiseLike<T> | void): value is Promise<T> {
-  return !!value && typeof value.then === 'function';
+export function isPromiseLike<T = any>(value?: unknown): value is PromiseLike<T> {
```

Comment:
It's a obviously a tradeoff, but I'd consider doing

```suggestion
export function isPromiseLike<T = unknown>(value?: unknown): value is PromiseLike<T> {
```

instead so one doesn't potentially introduce a hidden `any` by default each time they're using this function

Reply:
Sounds good, changed it :)